### PR TITLE
Simplify Zsh config and remove bloat

### PR DESCRIPTION
**Pull request by fesowowako**
# Pull Request

## Description

In my [feature request](https://github.com/prasanthrangan/hyprdots/issues/1880), I initially misstated that Powerlevel10k was unmaintained. However, recent activity shows it is still receiving updates.

- Revised Zsh Configuration: Improved clarity and ease of user configuration, maintaining the same functionality.
- Removed Dependencies: Eliminated the need for additional packages like Powerlevel10k and Oh My Zsh, simplifying the setup.
- Plugin Management: Utilised Zinit as the plugin manager and added three previously used plugins.
- Omitted Oh My Zsh: Chose not to include Oh My Zsh due to its bloat and noticeable impact on shell startup time.
- Implemented Zinit Snippets: Added plugins using Zinit’s snippet functionality for installation via URL.

Additional Notes:
- The maintainer of these dotfiles may need to review restore_shl.sh, as I’m uncertain about what to remove to prevent potential issues.

## Type of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] **Documentation update** (non-breaking change; modified files are limited to the documentations)
- [x] **Technical debt** (a code change that does not fix a bug or add a feature but makes something clearer for devs)
- [ ] **Other** (provide details below)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/prasanthrangan/hyprdots/blob/main/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My commit message follows the [commit guidelines](https://github.com/prasanthrangan/hyprdots/blob/main/CONTRIBUTING.md#git-commit-messages).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added a changelog entry.
- [x] I have added necessary comments/documentation to my code.
- [ ] I have added tests to cover my changes.
- [x] I have tested my code locally and it works as expected.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by rubiin** [comment]
I do agree on the bloat of ohmyzsh. This will be a breaking change
also I recently came across  this benchmark and zinit seems to be slower than the competitions https://github.com/rossmacarthur/zsh-plugin-manager-benchmark

File: Configs/.zshrc [hunk 1]
```diff
@@ -1,12 +1,84 @@
...
+setopt appendhistory
+setopt sharehistory
+setopt hist_ignore_space
+setopt hist_ignore_all_dups
+setopt hist_save_no_dups
+setopt hist_ignore_dups
+setopt hist_find_no_dups
+
+# Completion styling
+zstyle ':completion:*' matcher-list 'm:{a-z}={A-Za-z}'
```
Comment (zeerayne) [inline Configs/.zshrc:54]: Maybe matching regex `'m:{a-zA-Z}={A-Za-z}'` will be more flexible here?

**Comment by zeerayne** [comment]
Also, as an conversation: maybe to change p10k to i.e. oh-my-posh? Because p10k won't be receiving updates anymore.

**Comment by rubiin** [comment]
> Also, as an conversation: maybe to change p10k to i.e. oh-my-posh? Because p10k won't be receiving updates anymore.

It will be receiving updates but the pace of updates will be very slow . Its a matter of time , before a fork appears

**Comment by rubiin** [comment]
Yeah . Lets hold on with the PR and look for a viable option that doesnt compromise perf with ux like ohmyzsh

**Comment by zeerayne** [comment]
I'm using `sheldon` for plugins and `oh-my-posh` for theming. Can't complain about any performance issues on old i7 8700. Profile loads rapidly.
```sh
$ time zsh -i -c exit
zsh -i -c exit  0.04s user 0.03s system 91% cpu 0.081 total
```

**Comment by rubiin** [comment]
@zeerayne true. sheldon seems nice. I am also looking at zim

**Comment by zeerayne** [comment]
@rubiin Here you go

~/.config/sheldon/plugins.toml
```toml
shell = "zsh"

plugins.'zsh-users/zsh-completions'.github = 'zsh-users/zsh-completions'
plugins.'zsh-users/zsh-syntax-highlighting'.github = 'zsh-users/zsh-syntax-highlighting'
plugins.'zsh-users/zsh-history-substring-search'.github = 'zsh-users/zsh-history-substring-search'
plugins.'Aloxaf/fzf-tab'.github = 'Aloxaf/fzf-tab'
```

~/.config/oh-my-posh/config.toml
```toml
"$schema" = "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json"
version = 2
disable_notice = true

[[blocks]]
  type = 'prompt'
  alignment = 'left'
  newline = true

  [[blocks.segments]]
    type = 'path'
    style = 'plain'
    template = '{{ .Path }} '
    foreground = 'lightGreen'

    [blocks.segments.properties]
      style = 'full'

  [[blocks.segments]]
    type = 'git'
    style = 'plain'
    template = '<#ffffff>on</> {{ .HEAD }}{{ if gt .StashCount 0 }}  {{ .StashCount }}{{ end }} '

    [blocks.segments.properties]
      branch_icon = ''
      fetch_stash_count = true

  [[blocks.segments]]
    type = 'python'
    style = 'powerline'
    template = '[ {{ if .Error }}{{ .Error }}{{ else }}{{ if .Venv }}{{ .Venv }} {{ end }}{{ .Full }}{{ end }}] '
    foreground = '#906cff'

  [[blocks.segments]]
    type = 'go'
    style = 'powerline'
    template = '[ {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }}] '
    foreground = '#7FD5EA'

  [[blocks.segments]]
    type = 'node'
    style = 'powerline'
    template = '[ {{ if .PackageManagerIcon }}{{ .PackageManagerIcon }} {{ end }}{{ .Full }}] '
    foreground = '#76b367'

  [[blocks.segments]]
    type = 'ruby'
    style = 'powerline'
    template = '[ {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }}] '
    foreground = '#f44336'

  [[blocks.segments]]
    type = 'java'
    style = 'powerline'
    template = '[ {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }}] '
    foreground = '#ea2d2e'

  [[blocks.segments]]
    type = 'julia'
    style = 'powerline'
    template = '[ {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }}] '
    foreground = '#4063D8'

[[blocks]]
  type = 'rprompt'
  overflow = 'hidden'

  [[blocks.segments]]
    type = 'executiontime'
    style = 'plain'
    template = '{{ .FormattedMs }}'
    background = 'transparent'
    foreground = 'yellow'

    [blocks.segments.properties]
      threshold = 5000.0

[[blocks]]
  type = 'prompt'
  alignment = 'left'
  newline = true

  [[blocks.segments]]
    type = 'text'
    style = 'plain'
    foreground_templates = ['{{if gt .Code 0}}red{{end}}', '{{if eq .Code 0}}magenta{{end}}']
    template = '❯ '
    background = 'transparent'

[transient_prompt]
  foreground_templates = ['{{if gt .Code 0}}red{{end}}', '{{if eq .Code 0}}magenta{{end}}']
  template = '❯ '
  background = 'transparent'

[secondary_prompt]
  template = '❯❯ '
  background = 'transparent'
  foreground = 'magenta'
```

~/.zshrc
```zsh
#!/usr/bin/env zsh

# Environment variables
export EDITOR=vim

# If one of the VISUAL or EDITOR environment variables contain the string 'vi' when the shell starts up then it will be viins, 
# otherwise it will be emacs. bindkey's -e and -v options provide a convenient way to override this default choice.
bindkey -e

eval "$(dircolors -b)"
eval "$(oh-my-posh init zsh --config ~/.config/oh-my-posh/config.toml)"

# Load zsh plugins via sheldon
eval "$(sheldon source)"

# Check for zsh plugins updates weekly
SHELDON_CONFIG="${XDG_CONFIG_HOME:-${HOME}/.config}/sheldon"
if { [ ! -f "${SHELDON_CONFIG}/update.check" ] || [[ $(find "${SHELDON_CONFIG}/update.check" -mmin +10080) ]]; }; then
   sheldon lock --update
   touch "${SHELDON_CONFIG}/update.check"
fi

# History
HISTSIZE=10000
SAVEHIST=$HISTSIZE
HISTFILE=~/.zsh_history
HISTDUP=erase
setopt appendhistory
setopt sharehistory
setopt hist_ignore_space
setopt hist_ignore_all_dups
setopt hist_save_no_dups
setopt hist_ignore_dups
setopt hist_find_no_dups

zstyle ':completion:*' list-colors ${(s.:.)LS_COLORS}
zstyle ':completion:*' matcher-list 'm:{a-zA-Z}={A-Za-z}' 'r:|[._-]=* r:|=* l:|=*'
zstyle ':completion:*' menu no

zstyle ':completion:*:kill:*:processes' list-colors '=(#b) #([0-9]#)*=0=01;31'
zstyle ':completion:*:kill:*' command 'ps -u $USER -o pid,%cpu,tty,cputime,cmd'

# Load completions
autoload -Uz compinit && compinit

# Autocompletion using arrow keys (based on history)
## Arrow up
bindkey "${terminfo[kcuu1]}"  history-substring-search-up
bindkey "^[[A"                history-substring-search-up
## Arrow down
bindkey "${terminfo[kcud1]}"  history-substring-search-down
bindkey "^[[B"                history-substring-search-down

# Basic keybindings
## Home
bindkey  "${terminfo[khome]}" beginning-of-line
bindkey "^[[H"                beginning-of-line
## End
bindkey  "${terminfo[kend]}"  end-of-line
bindkey "^[[F"                end-of-line
## Del
bindkey  "${terminfo[kdch1]}" delete-char
bindkey "^[[3~"               delete-char

# Helpful aliases
alias  c='clear' # clear terminal
alias  l='eza -lh  --icons=auto' # long list
alias ls='eza -1   --icons=auto' # short list
alias ll='eza -lha --icons=auto --sort=name --group-directories-first' # long list all
alias ld='eza -lhD --icons=auto' # long list dirs

# Handy change dir shortcuts
alias ..='cd ..'
alias ...='cd ../..'
alias .3='cd ../../..'
alias .4='cd ../../../..'
alias .5='cd ../../../../..'

# Always mkdir a path (this doesn't inhibit functionality to make a single dir)
alias mkdir='mkdir -p'
```

**Comment by rubiin** [comment]
with sheldon 
```sh
zsh -i -c exit  0.08s user 0.05s system 108% cpu 0.115 total
```

**Comment by rubiin** [comment]
@fesowowako I suppose we can do that. Since powerlevel10k is on life support

**Comment by rubiin** [comment]
lets track for powerlevel10k on another PR. I am also playing with ohmyposh myself.

**Comment by rubiin** [comment]
I dug around ohmyzsh and it does set some stuffs  like https://github.com/ohmyzsh/ohmyzsh/blob/a72a26406ad3aa9a47c3f5227291bad23494bed0/lib/history.zsh#L42
and
https://github.com/ohmyzsh/ohmyzsh/blob/master/lib/completion.zsh

**Comment by kronikpillow** [comment]
I would have to disagree that zinit is the most "maintained and actively developed" ... as a matter of fact, there is so much contraversy going around zinit that it's insane, like ... the fact that the original maintainer deleted the entire zinit organization, then after lashback from community, and re-established forks, he reuploaded it as a new project, now you have multiple zinit repositories, with no clear line or sight, on which one is the official one, but either way, the newer one (one your not using) that is actually maintained by original dev, is not the one you presented in here ... althou im surprised by the benchmark results, zinit was once one of the fastest ...

zimfw is very nice, but lacks vi keybinding compatibility, most plugins and enrivonment settings are tailored towards Emacs like users of shells ^^

**Comment by kronikpillow** [comment]
I have a custom basic .zshrc no frameworks, and it loads much faster then any framework, while keeps most common used settings taken from zimfw, zap-zsh, and various other .zshrc's i found on the net

`zsh -i -c exit  0.03s user 0.01s system 99% cpu 0.043 total`

want me to share it?

**Comment by rubiin** [comment]
> zsh -i -c exit 0.03s user 0.01s system 99% cpu 0.043 total

sure go for it.

**Comment by rubiin** [comment]
@kronikpillow if you have a dotfiles repo, do share. Or maybe your zshrc

**Comment by T-Crypt** [comment]
@rubiin @fesowowako -- I will work on p10k alternative once I get through these PR's. I'm going to close this out for now, as this would be a breaking change and I'm not trying to have new users deal with that.

Please submit changes / fixes to the fix branch and not the main - I would like to keep main stable and have testing done on fixes before committing to main branch

**Pull request closed without merging**